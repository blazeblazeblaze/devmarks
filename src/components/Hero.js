import React from "react";

const getSubtitle = ({ title, subtitle }) => {
  if (subtitle !== undefined || title !== undefined) {
    return subtitle || `All about ${title}`;
  }
};

const Hero = props => {
  const { title } = props;

  return (
    <section className="hero">
      <div className="hero-body">
        <div className="container">
          <h1 className="title">{title}</h1>
          <h2>{getSubtitle(props)}</h2>
        </div>
      </div>
    </section>
  );
};

export default Hero;
