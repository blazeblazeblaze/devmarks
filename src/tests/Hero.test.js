import React from "react";
import Hero from "../components/Hero";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<Hero />", () => {
  const props = {
    title: "Custom title"
  };

  it("renders title on h1 level", () => {
    const tree = shallow(<Hero {...props} />);
    expect(tree.find("h1").text()).toBe(props.title);
  });

  it("renders subtitle on h2 level if provided", () => {
    const propsWithSubtitle = {
      ...props,
      subtitle: "subtitle"
    };
    const tree = shallow(<Hero {...propsWithSubtitle} />);

    expect(tree.find("h2").length).toBe(1);
  });

  it("renders subtitle on h2 level if provided", () => {
    const tree = shallow(<Hero {...props} />);

    expect(tree.find("h2").length).toBe(0);
  });

  it("matches snapshot", () => {
    const tree = shallow(<Hero {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
