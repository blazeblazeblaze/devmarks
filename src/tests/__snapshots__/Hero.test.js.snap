// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> matches snapshot 1`] = `
<section
  className="hero"
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Custom title
      </h1>
    </div>
  </div>
</section>
`;
